app/models/portfolio_item.rb

```diff
@@ -19,4 +24,10 @@ class PortfolioItem < ApplicationRecord
   def metadata
     {:user_capabilities => user_capabilities}
   end
+
+  private
+
+  def update_portfolio_stats
+    portfolio&.update_metadata
```

Comment:
Since the `:optional => true` is removed there is officially no valid way to have an empty `portfolio`. I actually think we wouldn't want that safety operator here if `portfolio` was nil. The callback should return an exception instead of quietly allowing nothing to happen.

Reply:
I quickly found out that the safe navigation was added because without it there are 10 test failures across to specs:

```
Failed examples:

rspec ./spec/requests/api/v1.0/portfolio_items_spec.rb
rspec ./spec/services/service_offering/add_to_portfolio_item_spec.rb
```

All but one are in `add_to_portfolio_item_spec`.  I'll see if I can address them here.  For now I left the safe navigation in place.